pkg/subnet/local.go

```diff
@@ -146,19 +153,65 @@ func (d *SubnetDeployer) doDeploy(chain string, chain_genesis string) error {
 	ctx := binutils.GetAsyncContext()
 
 	ux.Logger.PrintToUser("VM ready. Trying to boot network...")
-	info, err := cli.Start(
+	loadSnapshotOpts := []client.OpOption{
+		client.WithPluginDir(pluginDir),
+		client.WithExecPath(avalancheGoBinPath),
+	}
+	loadSnapshotsInfo, err := cli.LoadSnapshot(
+		ctx,
+		constants.DefaultSnapshotName,
+		loadSnapshotOpts...,
+	)
+	if err != nil {
+		if !strings.Contains(err.Error(), "already bootstrapped") {
+			return fmt.Errorf("failed to start network :%s", err)
+		}
+		ux.Logger.PrintToUser("Network has already been booted. Wait until healthy...")
+	} else {
+		ux.Logger.PrintToUser("Booting Network. Wait until healthy...")
+	}
+
+	d.log.Debug(loadSnapshotsInfo.String())
+
+	subnetIDs, blockchainsInfo, _, err := d.WaitForHealthy(ctx, cli, d.healthCheckInterval)
+	if err != nil {
+		_ = binutils.KillgRPCServerProcess()
+		return fmt.Errorf("failed to query network health: %s", err)
+	}
+
+	// choose next subnetID
+	sort.Strings(subnetIDs)
```

Comment:
So this is also something new here and I might just not understand.

Do we want to make all decisions for the user and deploy everything? What if that is not desired?
Also, why do have to do this in the first place? What is happening here? Aren't this subnets and blockchains supposed to be in the snapshot?

I would really like to see more documentation about what is going on here.

Reply:
The subnets IDs are in the snapshot but not the blockchains. The blockchain is going to be created here. We are selecting  (among the available ones in snapshot) a subnet ID to be used when creating the blockchain. 
It is pretty similar to version without snapshot but now the process of subnet id creation was done previously.
I don't understand on the deploy everything part of the comment

Reply:
adding this comment:
// in order to make subnet deploy faster, a set of validated subnet IDs is preloaded                                 // in the bootstrap snapshot                                                                                        
// we select one to be used for creating the next blockchain, for that we use the                                    // number of currently created blockchains as the index to select the next subnet ID,                               // so we get incremental selection

Reply:
OK. However, this only works for subnets where there are no controlKeys and thresholds to be observed...which might be ok for many situations but maybe not for all...

That's why I insisted that in the ANR PR we always allow to create subnets via normal path, and not via pre-creation always.